packages/slides-frontend/src/containers/Editor/SideBar/index.js

```diff
@@ -3,27 +3,58 @@ import {connect} from 'react-redux';
 import PropTypes from 'prop-types';
 
 import {Menu, Icon} from 'semantic-ui-react';
-import {addSlide, removeSlide, addItem} from '../../redux-store/DeckReducer/actions';
+import {addSlide, removeSlide, addItem, cloneSlide} from '../../redux-store/DeckReducer/actions';
+import {getCurrentSlide, getItems} from '../../redux-store/DeckReducer/selectors';
 import {uploadImageRequest} from '../../redux-store/PresentationReducer/actions';
-import {getCurrentSlide} from '../../redux-store/DeckReducer/selectors';
-import './index.css';
+import {getAssetsPath, getTitle} from '../../redux-store/PresentationReducer/selectors';
+import {deleteImage} from '../../../utils/requests';
 import {ItemTypes} from '../../redux-store/DeckReducer/definitions';
+import './index.css';
 // when i render SideBar onClick they will render something in the middle
 
-function SideBar({onAddSlide, onRemoveSlide, onAddText, currentSlide, onAddImage}) {
+function SideBar({
+  onAddSlide,
+  onRemoveSlide,
+  onAddText,
+  currentSlide,
+  onAddImage,
+  onCloneSlide,
+  itemsArray,
+  assetsPath,
+  username,
+  title,
+  token,
+  slidesStringified
+  }) {
+
   const addingSlide = () => {
     onAddSlide();
-    window.location = `#/${currentSlide + 1}`; // because slides here are starting from 1
+    window.location = `#/${currentSlide + 1}`;
   };
+
   const removingSlide = () => {
+    // check if pictures should be deleted from backend
+    itemsArray.map(item => {
```

Comment:
`itemsArray.forEach` - `map` is used when you want to return a result of the iteration